### ISSUE TITLE
Improved training pipeline

### ISSUE THREAD

**Issue by BerkIGuler** [BerkIGuler/WirelessContrastiveMaskedLearning#1]
- Add and test gradient accumulation feature to handle larger batch sizes with single GPU
- Add and test multi-GPU training
